File: 1046-last-stone-weight/1046-last-stone-weight.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> mh = new PriorityQueue<>(Collections.reverseOrder());

        for(int num : stones){
            mh.offer(num);
        }
        while(mh.size() > 1){
            int y = mh.poll();
            int x = mh.poll();

            if(x != y){
                mh.offer(y - x);
            }
        }
        if(mh.isEmpty()) return 0;
        else return mh.peek();
        
    }
}

File: 1046-last-stone-weight/1046-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: list[int]) -> int:
        while len(stones) > 1:
            stones.sort(reverse=True)
            y = stones.pop(0)
            x = stones.pop(0)
            
            if y != x:
                stones.append(y - x)

        return stones[0] if stones else 0
